iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -19,35 +19,69 @@
-  private static final String DISABLED_CERTIFICATE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String MISSING_CERTIFICATE_MESSAGE = "Omitting \"%s\" disables certificate-based authentication. Make sure it is safe here.";
+  private static final String DISABLED_CERTIFICATE_MESSAGE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String ALLOWING_NO_CERTIFICATE_MESSAGE = "Connections without client certificates will be permitted. Make sure it is safe here.";
+
+  private static final List<String> PATH_CONTAINER_APPS = ArmTreeUtils.computePath("configuration/ingress");
 
   @Override
   protected void registerResourceConsumer() {
-    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkNegotiateClientCertificate);
+    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkHostnameConfigurations);
+    register("Microsoft.App/containerApps", CertificateBasedAuthenticationCheck::checkContainerApps);
   }
 
-  private static void checkNegotiateClientCertificate(CheckContext ctx, ResourceDeclaration resource) {
-    PropertyUtils.value(resource, "negotiateClientCertificate")
+  private static void checkHostnameConfigurations(CheckContext ctx, ResourceDeclaration resource) {
+    PropertyUtils.get(resource, "negotiateClientCertificate")
       .ifPresentOrElse(
-        value -> {
-          if (isBooleanFalse(value)) {
-            ctx.reportIssue(value, DISABLED_CERTIFICATE);
+        property -> {
+          if (isBooleanFalse(property.value())) {
+            ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
           }
         },
-        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE, "negotiateClientCertificate")));
+        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE_MESSAGE, "negotiateClientCertificate")));
+  }
+
+  private static void checkContainerApps(CheckContext ctx, ResourceDeclaration resource) {
+    if (isResourceVersionBefore(resource, "2022-10-01")) {
+      List<Tree> resolvedTrees = ArmTreeUtils.resolveTreePath(resource, new LinkedList<>(PATH_CONTAINER_APPS));
+      for (Tree resolvedTree : resolvedTrees) {
+        PropertyUtils.get(resolvedTree, "clientCertificateMode")
+          .ifPresentOrElse(
+            property -> {
+              if (isStringValue(property.value(), "accept")) {
+                ctx.reportIssue(property, ALLOWING_NO_CERTIFICATE_MESSAGE);
+              } else if (isStringValue(property.value(), "ignore")) {
+                ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
+              }
+            },
+            () -> ctx.reportIssue(resolvedTree, String.format(MISSING_CERTIFICATE_MESSAGE, "clientCertificateMode")));
```

Comment:
Same here. It would be like this:
```
container.attribute("clientCertificateMode")
 .reportIf(isValue("accept"), ALLOWING_NO_CERTIFICATE_MESSAGE)
 .reportIf(isValue("ignore"), DISABLED_CERTIFICATE_MESSAGE)
 .reportIfAbsent(MISSING_CERTIFICATE_MESSAGE)
```